lib/vanagon/component.rb

```diff
@@ -231,13 +231,18 @@ def get_build_dir
     # Fetches secondary sources for the component. These are just dumped into the workdir currently.
     #
     # @param workdir [String] working directory to put the source into
-    def get_sources(workdir)
+    def get_sources(workdir) # rubocop:disable Metrics/AbcSize
+      # create an empty array if @extract_with doesn't already exist
+      @extract_with = Array(@extract_with)
```

Comment:
Using `[@extract_with]` would be a bit more idiomatic than `Array(@extract_with)`. Not a deal breaker, but worth pointing out. Also, use the accessor (`the #extract_with` method) instead of the instance variable (`@extract_with`).